### PR TITLE
Code cleanup and inline

### PR DESCRIPTION
**Pull request by HalidOdat**
It changes the following:
- Change `Context` variable name to `context` (instead of `engine` or `interpreter`)
- Added `#[inline]`
- change gc import to use gc module

### CODE REVIEW

File: boa/src/builtins/infinity/mod.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ impl BuiltIn for Infinity {
         Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::PERMANENT
     }
 
-    fn init(_context: &mut Context) -> (&'static str, Value, Attribute) {
+    fn init(_: &mut Context) -> (&'static str, Value, Attribute) {
```
Comment (Razican) [inline boa/src/builtins/infinity/mod.rs:28]: Why this change?

File: boa/src/builtins/nan/mod.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ impl BuiltIn for NaN {
         Attribute::READONLY | Attribute::NON_ENUMERABLE | Attribute::PERMANENT
     }
 
-    fn init(_context: &mut Context) -> (&'static str, Value, Attribute) {
+    fn init(_: &mut Context) -> (&'static str, Value, Attribute) {
```
Comment (Razican) [inline boa/src/builtins/nan/mod.rs:29]: Why this change?

**Review by Razican — COMMENTED** [review]
This looks good to me, but why was the test262 submodule changed? It looks like it was reverted to a version in September. Could we do a `git pull origin main` in there?

File: boa/src/builtins/infinity/mod.rs (hunk 1 above)
Comment (HalidOdat) [reply boa/src/builtins/infinity/mod.rs:28]: its because the context was not used to changed it to `_` like the other builtin functins

**Review by RageKnify — APPROVED** [review]

**Comment by RageKnify** [comment]
My bad, new `commit` should work ok.

**Review by Razican — APPROVED** [review]
